libraries/imgui-opencv-poco/src/gui_renderer.cpp

```diff
@@ -0,0 +1,152 @@
+#include "gui_renderer.hpp"
+
+static void glfw_error_callback(int error, const char *description)
+{
+    fprintf(stderr, "Glfw Error %d: %s\n", error, description);
+}
+
+GUIRenderer::GUIRenderer(void) : texture_id_(-1), threshold_(127)
+{
+}
+
+GUIRenderer::~GUIRenderer(void)
+{
+}
+
+int GUIRenderer::InitGUI()
+{
+    glfwSetErrorCallback(glfw_error_callback);
+    if (!glfwInit())
+        return 1;
+
+        // Decide GL+GLSL versions
+#if __APPLE__
+    // GL 3.2 + GLSL 150
+    const char *glsl_version = "#version 150";
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 2);
+    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE); // 3.2+ only
+    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);           // Required on Mac
+#else
+    // GL 3.0 + GLSL 130
+    const char *glsl_version = "#version 130";
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+    //glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);  // 3.2+ only
+    //glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);            // 3.0+ only
+#endif
+
+    // Create window with graphics context
+    window_ = glfwCreateWindow(1280, 720, "imgui-opencv demo", NULL, NULL);
+    if (window_ == NULL)
+        return 1;
+    glfwMakeContextCurrent(window_);
+    glfwSwapInterval(1); // Enable vsync
+
+    // Initialize OpenGL loader
+    bool err = glewInit() != GLEW_OK;
+    if (err)
+    {
+        fprintf(stderr, "Failed to initialize OpenGL loader!\n");
+        return 1;
+    }
+    // Setup Dear ImGui context
+    IMGUI_CHECKVERSION();
+    ImGui::CreateContext();
+    ImGuiIO &io = ImGui::GetIO();
+    (void)io;
+    ImGui::StyleColorsDark();
+    // Setup Platform/Renderer bindings
+    ImGui_ImplGlfw_InitForOpenGL(window_, true);
+    ImGui_ImplOpenGL3_Init(glsl_version);
+    return 0;
+}
+
+void GUIRenderer::UpdateTexture()
+{
+    unsigned char *data = thresholded_image_.ptr();
+    image_width_ = thresholded_image_.cols;
+    image_height_ = thresholded_image_.rows;
+    if (texture_id_ == -1)
+        glGenTextures(1, &texture_id_);
+    glBindTexture(GL_TEXTURE_2D, texture_id_);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    glPixelStorei(GL_UNPACK_ROW_LENGTH, 0);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, image_width_, image_height_, 0, GL_BGR, GL_UNSIGNED_BYTE, data);
+    glGenerateMipmap(GL_TEXTURE_2D);
+}
+
+void GUIRenderer::ShowImage()
+{
+    ImGui::Begin("Image");
+    // Image downloaded from: https://www.pexels.com/photo/green-bird-1661179/
+    static char image_url[256] = "https://raw.githubusercontent.com/czoido/imgui-opencv/master/data/bird.jpeg";
```

Comment:
I think we can move this bird pic too to Conan examples.

Reply:
Lets do it when the PR gets merged and we get a real valid URL?

Reply:
Nevermind, the file was there, fixed.